src/Android/Avalonia.Android/Platform/AndroidNativeControlHostImpl.cs

```diff
@@ -0,0 +1,187 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Android.Views;
+using Android.Widget;
+
+using Avalonia.Controls.Platform;
+using Avalonia.Platform;
+
+namespace Avalonia.Android.Platform
+{
+    internal class AndroidNativeControlHostImpl : INativeControlHostImpl
+    {
+        private readonly AvaloniaView _avaloniaView;
+
+        public AndroidNativeControlHostImpl(AvaloniaView avaloniaView)
+        {
+            _avaloniaView = avaloniaView;
+        }
+
+        public INativeControlHostDestroyableControlHandle CreateDefaultChild(IPlatformHandle parent)
+        {
+            return new AndroidViewControlHandle(new FrameLayout(_avaloniaView.Context!), false);
+        }
+
+        public INativeControlHostControlTopLevelAttachment CreateNewAttachment(Func<IPlatformHandle, IPlatformHandle> create)
+        {
+            var holder = new AndroidViewControlHandle(_avaloniaView, false);
+            AndroidNativeControlAttachment? attachment = null;
+            try
+            {
+                var child = create(holder);
+                // It has to be assigned to the variable before property setter is called so we dispose it on exception
+#pragma warning disable IDE0017 // Simplify object initialization
+                attachment = new AndroidNativeControlAttachment(child);
+#pragma warning restore IDE0017 // Simplify object initialization
+                attachment.AttachedTo = this;
+                return attachment;
+            }
+            catch
+            {
+                attachment?.Dispose();
+                holder?.Destroy();
+                throw;
+            }
+        }
+
+        public INativeControlHostControlTopLevelAttachment CreateNewAttachment(IPlatformHandle handle)
+        {
+            return new AndroidNativeControlAttachment(handle)
+            {
+                AttachedTo = this
+            };
+        }
+
+        public bool IsCompatibleWith(IPlatformHandle handle) => handle.HandleDescriptor == AndroidViewControlHandle.AndroidDescriptor;
+
+        class AndroidNativeControlAttachment : INativeControlHostControlTopLevelAttachment
+        {
+            // ReSharper disable once NotAccessedField.Local (keep GC reference)
+            private IPlatformHandle? _child;
```

Comment:
Is GC reference required here? We are referencing the view anyway.

Reply:
I was keeping in mind cases when developer has some exotic IPlatformHandle implementation, that needs to be kept alive for a whole lifetime of the object.
But:
1. NativeControlHost control itself already keeps this reference
2. That's very unlikely to be the case

So, yeah, removing it from here and ios impl